nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -176,13 +176,25 @@ protected void flushKuduSession(final KuduSession kuduSession, boolean close, fi
         }
     }
 
-    protected KerberosUser loginKerberosKeytabUser(final String principal, final String keytab) throws LoginException {
-        final KerberosUser kerberosUser = new KerberosKeytabUser(principal, keytab);
+    protected KerberosUser loginKerberosKeytabUser(final String principal, final String keytab, ProcessContext context) throws LoginException {
+        final KerberosUser kerberosUser = new KerberosKeytabUser(principal, keytab) {
+            @Override
+            public synchronized boolean checkTGTAndRelogin() throws LoginException {
+                boolean didRelogin = super.checkTGTAndRelogin();
+
+                if (didRelogin) {
+                    final KerberosAction<KuduClient> kerberosAction = new KerberosAction<>(this, () -> buildClient(context), getLogger());
+                    AbstractKuduProcessor.this.kuduClient = kerberosAction.execute();
+                }
+
+                return didRelogin;
+            }
+        };
         kerberosUser.login();
         return kerberosUser;
     }
 
-    protected KerberosUser loginKerberosPasswordUser(final String principal, final String password) throws LoginException {
+    protected KerberosUser loginKerberosPasswordUser(final String principal, final String password, ProcessContext context) throws LoginException {
```

Comment:
The context parameter is unused here. Should not we execute the same logic as in `loginKerberosKeytabUser()`?

Reply:
It looks used in `buildClient(context)` below.

Reply:
Yes, it is used now. It was an earlier comment, has been resolved in the meantime, just forgot to mark it resolved.